src/actions/referralsActions.js

```diff
@@ -147,63 +152,62 @@ export const sendReferralInvitationsAction = (invitationContacts: ReferralContac
       const { email, phone } = invitation;
       const token = await api.generateReferralToken(walletId);
 
-      if (token.result === 'success') {
-        const referralLink = await getUserReferralLink(walletId, {
-          email,
-          phone,
-          token: token.token,
-        });
-
-        const { error } = await api.sendReferralInvitation({
-          token: token.token,
-          walletId,
-          referralLink,
-          email,
-          phone,
-        });
-
-        if (error) {
-          const errorMessage = get(error, 'response.data.message');
-          inviteErrorAction(dispatch, errorMessage);
-        } else {
-          let updatedInvitationCount = count + invitationContacts.length;
-          const currentDate = formatDate(new Date(), 'YYYY-MM-DD');
-          if (date !== currentDate) updatedInvitationCount = invitationContacts.length;
-          const updatedAlreadyInvitedContacts = [...alreadyInvitedContacts, ...invitationContacts];
-          await dispatch(saveDbAction('referralData', {
-            referrals: {
-              alreadyInvitedContacts: updatedAlreadyInvitedContacts,
-              sentInvitationsCount: { count: updatedInvitationCount, date: currentDate },
-            },
-          }));
-          inviteSentAction(dispatch, {
-            alreadyInvitedContacts: invitationContacts,
-            sentInvitationsCount: { count: updatedInvitationCount, date: currentDate },
-          });
-        }
-      } else {
+      if (token.result !== 'success') {
         inviteErrorAction(dispatch);
+        return;
+      }
+
+      const referralLink = await getUserReferralLink(walletId, {
+        email,
+        phone,
+        token: token.token,
+      });
+
+      const { error } = await api.sendReferralInvitation({
+        token: token.token,
+        walletId,
+        referralLink,
+        email,
+        phone,
+      });
+
+      if (error) {
+        const errorMessage = get(error, 'response.data.message');
+        inviteErrorAction(dispatch, errorMessage);
+        return;
+      }
+
+      let updatedInvitationCount = count + invitationContacts.length;
+      const currentDate = formatDate(new Date(), 'YYYY-MM-DD');
+      if (date !== currentDate) {
+        updatedInvitationCount = invitationContacts.length;
       }
+      inviteSentAction(dispatch, {
```

Comment:
This looks a bit weird. All actions have dispatcher within them, idk if this action can be built the way others are?